x/mongo/driver/topology/topology.go

```diff
@@ -612,10 +610,16 @@ func (t *Topology) apply(ctx context.Context, desc description.Server) descripti
 	}
 
 	prev := t.fsm.Topology
-
+	ind, ok := t.fsm.findServer(desc.Addr)
```

Comment:
If we have this check, do we still need the check on `t.servers[desc.Addr]` above? Seems like we can just have:

```
ind, ok := t.fsm.findServer(desc.Addr)
if !ok {
    return desc
}
```

Reply:
Sorry if this wasn't clear, but I was trying to say that we should keep the `findServer` check and get rid of the `t.servers` one. `fsm.findServer` works off of the current topology description, which is more in line with the spec pseudocode.